src/lib/output/themes/default/assets/typedoc/Hierarchy.ts

```diff
@@ -0,0 +1,339 @@
+        branch.parents?.sort((a, b) => a.text.localeCompare(b.text));
+
+        branch.children?.sort((a, b) => a.text.localeCompare(b.text));
+    });
+
+    const reverseTree: HierarchyElement[] = [];
+
+    const visited: Set<HierarchyElement> = new Set();
+
+    function visit(branch: HierarchyElement) {
+        visited.add(branch);
+
+        branch.children?.forEach(
+            (child) => !visited.has(child) && visit(child),
+        );
+
+        reverseTree.push(branch);
+    }
+
+    tree.forEach((branch) => !visited.has(branch) && visit(branch));
+
+    return reverseTree.reverse();
+}
+
+function growBranches(tree: HierarchyElement[]) {
+    tree = tree.slice();
+
+    const roots = tree.filter((branch) => !branch.parents?.length);
+
+    const minDepth = Math.min(...roots.map((branch) => branch.depth!));
+
+    roots.forEach((root) => {
+        while (root.depth! > minDepth) {
+            const newRoot = {
+                html: "",
+                text: "",
+                children: [root],
+                class: "tsd-hierarchy-spacer",
+                depth: root.depth! - 1,
+            };
+
+            root.parents = [newRoot];
+
+            tree.splice(tree.indexOf(root), 0, newRoot);
+
+            root = newRoot;
+        }
+    });
+
+    return tree;
+}
+
+function buildList(
+    branches: HierarchyElement[],
+    baseUrl: string,
+    summary: boolean,
+    reverse: boolean = true,
+    renderedBranches: Set<HierarchyElement> = new Set(),
+) {
+    if (branches.every((branch) => renderedBranches.has(branch)))
+        return undefined;
+
+    const list = document.createElement("ul");
+    list.classList.add("tsd-hierarchy");
+
+    if (reverse) branches = branches.slice().reverse();
+
+    branches.forEach((currentBranch) => {
+        if (renderedBranches.has(currentBranch)) return;
+
+        renderedBranches.add(currentBranch);
+
+        const item = document.createElement("li");
+
+        if (reverse) list.prepend(item);
+        else list.append(item);
+
+        item.className = currentBranch.class;
+
+        if (currentBranch.text) {
+            if (summary) {
+                item.innerHTML += `<a id="${currentBranch.text}" class="tsd-anchor"></a>`;
+
+                const anchor = document.createElement("a");
+                anchor.href = baseUrl + currentBranch.path!;
+                item.append(anchor);
+
+                anchor.innerHTML += `<svg width="20" height="20" viewBox="0 0 24 24" fill="none" class="tsd-kind-icon"><use href="#icon-${currentBranch.kind!}"></use></svg>`;
+                anchor.innerHTML += `<span>${currentBranch.text}</span>`;
+            } else if (item.classList.contains("tsd-hierarchy-target")) {
+                item.innerHTML += `<span>${currentBranch.text}</span>`;
+            } else {
+                item.innerHTML += currentBranch.html.replace(
+                    /(?<=<a [^>]*(?<=\s)href=")(?![a-zA-Z]+:\/\/)([^"]*)(?="[^>]*>)/g,
```

Comment:
That's... no... that's not going to be okay. It'd be better to set the innerHTML, then query for anchor elements and update the href attribute.

Reply:
That's totally fair and an easy fix. I might be too used to abusing regex, I didn't even think about it.